### PR TITLE
[onert] Introduce LoweredTrainableGraph

### PR DESCRIPTION
**Pull request by ragmani**
This commit introduces LoweredTrainableGraph lowering TrainableGraph.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: runtime/onert/core/src/compiler/train/LoweredTrainableGraph.cc [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+#include "../pass/ConstantLoweringPass.h"
+#include "../pass/PassRunner.h"
+#include "../pass/PermutationEliminationPass.h"
+#include "../pass/PermutationInsertionPass.h"
+#include "../pass/PermutationOperationPass.h"
+#include "../../backend/builtin/Config.h"
+#include "../../dumper/text/GraphDumper.h"
+#include "../../ir/verifier/Verifier.h"
+// TODO Enable after merging TrainableOperationConverter
+// #include "TrainableOperationConverter.h"
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/train/LoweredTrainableGraph.cc:30]: I'm going to remove this commented out after merging `TrainableOperationConverter`.

File: runtime/onert/core/src/compiler/train/LoweredTrainableGraph.cc [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+  // _trainable_graph.operations().iterate(
+  //   [&](const onert::ir::OperationIndex &index, const onert::ir::IOperation &op) {
+  //     if (op.opcode() == ir::OpCode::Permute)
+  //     {
+  //       auto top = op_converter(index);
+  //       auto gen_index = _trainable_graph.replaceOperation(index, std::move(top));
+  //       UNUSED_RELEASE(gen_index);
+  //       assert(gen_index == index);
+  //     }
+  //   });
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/train/LoweredTrainableGraph.cc:120]: ditto

**Review by jyoungyun — APPROVED** [review]
LGTM

File: runtime/onert/core/src/dumper/text/GraphDumper.cc [hunk 3]
```diff
@@ -69,13 +69,20 @@ std::string formatOperation(const ir::Graph &graph, ir::OperationIndex ind)
   return ss.str();
 }
 
+std::string formatOperation(const ir::Graph &graph, ir::OperationIndex ind)
+{
+  std::stringstream ss;
+  const auto &op = graph.operations().at(ind);
+  return formatOperation(op, ind);
+}
```
Comment (glistening) [inline runtime/onert/core/src/dumper/text/GraphDumper.cc:77]:
```suggestion

}
```

**Comment by glistening** [comment]
@nnfw-bot test onert-x64-release

**Review by glistening — APPROVED** [review]
LGTM

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**